### DISCUSSION TITLE
Automatically add publicly available dividend payments to activity list

### DISCUSSION THREAD

**Discussion by cheesestickstomatojuice** [ghostfolio/ghostfolio#3104] [Ideas]
I would like to see a feature in ghostfolio that allows automatic tracking of dividends paid by the stocks I hold. Current manual entry methods take too much time and lead to human error. Investing.com and Nasdaq both offer a dividend calender, maybe you can do something with that? Manual entry of dividend transactions is tedious and prone to error especially if you hold a lot of stocks that have dividends. An automatic solution would significantly increase usability and convenience while ensuring accurate recordkeeping.

**Comment by cjboyle** [comment]
One issue I foresee with any automated functionality for this is whether or not a dividend paid to foreign investors is taxed before it is paid out.

For example, in Canada, we "pay" a 15% Withholding Tax on U.S. dividend distributions but not on return-of-capital distributions (by "pay", I mean the tax is taken out of the distribution payment directly). The tax is also handled differently based on the type of the account: for tax-free accounts, it is 15% full stop; for tax-differed/retirement accounts, dividends are usually *not* taxed at all; with personal taxable accounts, the tax is recoverable in part or in full on federal tax returns.

In short, if this is implemented it should at least have a switch to make it optional.

**Comment by pedrommvv** [comment]
I think you can import dividends by each stock you hold in ghostfolio, and you can select all or some of them in the generated list. I think it does works well. 

Regarding withholding tax for dividends, I already mentioned on another topic, this is really important... It would be like the `fee` field, but it deducted from total dividend returns and statistics

**Comment by cheesestickstomatojuice** [comment]
If automatic imports are an issue due to varying tax rules across countries and potential complications, adding an dividend calendar might be more convenient for tracking with the stocks you hold and their corresponding dividends, which can then be added manually along with applicable taxes.

**Reply by pedrommvv** [reply]
I don't think having automatic imports are the problem. There's multiple solutions possible but not enough programmers... That's why for now, I suggest only a percentage input field without any calculations, save it anywhere in the DB, for later implementation.
